### ISSUE TITLE
about ViT-SO400M-14-SigLIP-384 model

### ISSUE THREAD

**Issue by ctgushiwei** [mlfoundations/open_clip#1056]
I use the follow code to load the ViT-so400M-siglip-384,The following error will occur

![Image](https://github.com/user-attachments/assets/c6c8facb-44d0-40f1-ac77-96c90f6dd0d4)
model_path = "../pretrain_weights/ViT-SO400M-14-SigLIP-384/open_clip_pytorch_model.bin"
model_name = "ViT-SO400M-14-SigLIP-384"

model, _, preprocess = open_clip.create_model_and_transforms(model_name = model_name, pretrained = model_path)
tokenizer = open_clip.get_tokenizer(model_name)

**Comment by rwightman** [comment]
@ctgushiwei models that rely on a huggingface tokenizer like SigLIP won't work completely offline like that, you need to first download the model while connected then use cache, you can use the `cache_dir` arg to force the download / reload to a known folder. 

It would be possible to add a fully folder based offline load but it's a bit of work and there are a number of details to get right, some have tried but it's always wrong and causes regressions for other areas (see eg #1053). Hasn't been a priority because most users can at least do the online cache before moving to an offline state.

**Comment by ctgushiwei** [comment]
this issue have solved。but when I use the follow code to infer the model of ViT-SO400M-16-SigLIP2-384，Another issue has occurred, which seems to be related to tokenizer. 

![Image](https://github.com/user-attachments/assets/a0aaabf4-72c9-4fc9-8780-0867482dd7d9)

The code is as follows:
import torch
import torch.nn.functional as F
from urllib.request import urlopen
from PIL import Image
from open_clip import create_model_from_pretrained, get_tokenizer # works on open-clip-torch >= 2.31.0, timm >= 1.0.15

model, preprocess = create_model_from_pretrained('hf-hub:timm/ViT-SO400M-16-SigLIP2-384')
tokenizer = get_tokenizer('hf-hub:timm/ViT-SO400M-16-SigLIP2-384')

image = Image.open(urlopen(
'https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/beignets-task-guide.png'
))
image = preprocess(image).unsqueeze(0)

labels_list = ["a dog", "a cat", "a donut", "a beignet"]
text = tokenizer(labels_list, context_length=model.context_length)

with torch.no_grad(), torch.cuda.amp.autocast():
image_features = model.encode_image(image, normalize=True)
text_features = model.encode_text(text, normalize=True)
text_probs = torch.sigmoid(image_features @ text_features.T * model.logit_scale.exp() + model.logit_bias)

zipped_list = list(zip(labels_list, [100 * round(p.item(), 3) for p in text_probs[0]]))
print("Label probabilities: ", zipped_list)

**Comment by rwightman** [comment]
@ctgushiwei is your transformers version at the latest? It needs to be relatively recent to load that tokenizer.
